gprofiler/profilers/dotnet.py

```diff
@@ -0,0 +1,157 @@
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+import functools
+import os
+import signal
+import subprocess
+from pathlib import Path
+
+from threading import Event
+from typing import Any, Dict, List, Optional
+from granulate_utils.linux.process import process_exe
+from psutil import Process
+from gprofiler.exceptions import ProcessStoppedException, StopEventSetException
+from gprofiler.gprofiler_types import ProfileData
+from gprofiler.metadata.application_metadata import ApplicationMetadata
+from gprofiler.log import get_logger_adapter
+from gprofiler.profilers.profiler_base import ProcessProfilerBase
+from gprofiler.profilers.registry import register_profiler
+from gprofiler.utils import pgrep_maps, random_prefix, removed_path, resource_path, run_process
+from gprofiler.utils.speedscope import load_speedscope_as_collapsed
+logger = get_logger_adapter(__name__)
+
+
+class DotnetMetadata(ApplicationMetadata):
+
+    _DOTNET_VERSION_TIMEOUT = 3
+
+    @functools.lru_cache(4096)
+    def _get_dotnet_version(self, process: Process) -> str:
+        if not os.path.basename(process_exe(process)).startswith("dotnet"):
+            # TODO: for dynamic executables, find the ruby binary that works with the loaded libruby, and
+            # check it instead. For static executables embedding libruby - :shrug:
```

Comment:
Comment is irrelevant here (it's for ruby).

I'm actually not sure if it's relevant for dotnet - i.e if a program can be named `abc` and have `libdotnet.so` loaded like sometimes happens for java/python.

You can remove the comment but keep the logic raising `NotImplementedError`, incase